repo/blob/s3/s3_options.go

```diff
@@ -21,7 +21,14 @@ type Options struct {
 
 	AccessKeyID     string `json:"accessKeyID"`
 	SecretAccessKey string `json:"secretAccessKey" kopia:"sensitive"`
-	SessionToken    string `json:"sessionToken"    kopia:"sensitive"`
+	SessionToken    string `json:"sessionToken" kopia:"sensitive"`
+
+	// Used for assume role authentication.
+	RoleARN      string        `json:"roleARN"`
+	SessionName  string        `json:"sessionName"`
+	RoleDuration time.Duration `json:"duration"`
```

Comment:
While this marshals correctly, it does not "roundtrip" (i.e., unmarshal back).
We can merge this, and iterate quickly to change the type to `jsonencoding.Duration`